src/docs/cookie-clicker-swiftui.md

```diff
@@ -0,0 +1,314 @@
+
+![step2](assets/cookie_clicker_swiftui/step2.png)
+
+### 3. Fill in the project details:
+
+- Product Name: Clicker
+- Team: None
+- Organization Name: Whatever you want — your name is always a good fill-in
+- Organization Identifier: com.(OrganizationName)
+
+Make sure Swift is selected as the language.
+  
+Make sure SwiftUI is selected as the User Interface.
+
+Untick the 3 boxes at the bottom.
+
+Click Next
+
+![step3](assets/cookie_clicker_swiftui/step3.png)
+
+### 4. Select a sensible place to save your project, then hit **Create**.
+
+Take some time to look around the project that's been created with your coach. What files are there? How do you run your app?
+  
+If you are running Mac OS X Catalina, try running the preview to see what the default project provides.
+
+## Add a Button
+
+### 5. Open up the **Navigation Area** in Xcode, and then the `ContentView.swift` file and its associated Canvas.
+
+![step4](assets/cookie_clicker_swiftui/step4.gif)
+  
+If you click on the `Resume` button in the top right, and you are running Mac OS X Catalina you will see a preview of the **Hello World** application that the template has provided for you.
+  
+### 6. Edit the Canvas and add a Button
+
+When using SwiftUI the Canvas provides a live view of your code. Editing the Canvas will also update your code! We shall be adding code from the Editor a little later on, but for now just add a button to the canvas.
+  
+- Open the Library
+- Make sure you are browsing the Views
+- Search for Button
+- Drag the button under the label
+
+### ![step5](assets/cookie_clicker_swiftui/step5.gif)
+
+If you look at the Editor, you can see that this has created a _VStack_ with both the label and the button. A _VStack** is a built in view that arranges the contained views vertically.
+
+### 7. Run the app on the Simulator or the canvas.
+
+You can either run the app in the simulator or use the live preview in the Canvas to see the visual effect of pressing the button.
+ 
+Celebrate appropriately! 
+
+Try changing the `VStack` to an `HStack` Are the results what you expect?
+
+However, it would be even more awesome if our button actually did something.
+
+## Make the Button do something
+
+To do this we are going to need some state in our view. SwiftUI is a _declarative_ framework. That means that you write the view to show different things according to its state, and actions change the state. Internally, SwiftUI keeps track of the state, and when it changes it redraws the view.
+
+### 8. Create some state for the view
+
+Create a state variable at the top of the definition of `ContentView`. We make it `private` because it is internal to the view, and we give it a default value of `false` which implicitly makes `showGreeting` a variable that holds a `Bool` (Now is the time to ask your Coach questions).
+
+```swift
+struct ContentView: View {
+    @State private var showGreeting = false
```

Comment:
is there any way we can do this without using @state variables? e.g. create a standard Int variable that holds how many times you've tapped the button. And then in the next section hook up the button to a regular method that increments (and displays?) that counter?

Reply:
Maybe it's possible, but I wouldn't want to go that way. SwiftUI depends very much on the `@State`. It's what makes it a declarative way of writing views. It's not just `@State` (which is internal to a view) but also `@ObservedObject` `@Environment` and `@EnvironmentObject` which make this all much more powerful and I was hoping to cover those in subsequent tutorials - (I have a plan 😄 ).

Another reason I'd rather not change this is because If we are going to coach attendees on SwiftUI, I think it's better to teach them real world code, however simple, that's the benefit of having coaches, rather than throwing a tutorial over the wall. There are many tutorials about this, but I like that Codebar does this with a coach, who can explain things further from their own experience.

Does my reticence make sense in this context?

Reply:
Yeah! I think that definitely makes sense. 

Would you be able to explain what a @state variable is - or link to a place that explains well what it is? I love that we have 2:1 learning too, but some people use our tutorials at home on their own, or in different group settings too.